bundles/org.openhab.transform.basicprofiles/src/test/java/org/openhab/transform/basicprofiles/internal/profiles/StateFilterProfileTest.java

```diff
@@ -690,6 +692,14 @@ public static Stream<Arguments> testFunctions() {
                 Arguments.of(decimalItem, "$DELTA >= 1", decimals, DecimalType.valueOf("10"), true), //
                 Arguments.of(decimalItem, "$DELTA >= 1", decimals, DecimalType.valueOf("5.5"), false), //
 
+                // Test for the use of relative unit in DELTA checks
+                Arguments.of(temperatureItem, "$DELTA > 0.2 °F", celsiusQuantities, QuantityType.valueOf("5.1 °C"),
+                        false), //
+                Arguments.of(temperatureItem, "0.2 °F < $DELTA", celsiusQuantities, QuantityType.valueOf("5.1 °C"),
+                        false), //
+                Arguments.of(temperatureItem, "$DELTA < 2 °F", celsiusQuantities, QuantityType.valueOf("6 °C"), true), //
+                Arguments.of(temperatureItem, "2 °F > $DELTA", celsiusQuantities, QuantityType.valueOf("6 °C"), true), //
+
```

Comment:
Suggest to add the following

- tests in Celsius
- tests in Kelvin

Reply:
added more tests and split these up into a separate test